### PR TITLE
Initial version of atom flatpak

### PR DESCRIPTION
**Pull request by krnowak**
https://phabricator.endlessm.com/T14295

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/bash
+
+set -e
+
+if [[ ! -d repo ]]
+then
+    ostree  init --mode=archive-z2 --repo=repo
```
Comment (lilic) [inline build.sh]: one space too many ;)

File: resources/install [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/bash
+
+set -e
+
+prefix=$(cat prefix)
+
+appdir="${prefix}/share/applications"
+olddesktop="${appdir}/atom.desktop"
```
Comment (lilic) [inline resources/install:8]: You can just use `rename-desktop-file` in your json manifest file.

> Any desktop file with this name will be renamed to a name based on id during the cleanup phase.

File: atom/install [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+set -e
+
+prefix=$(cat prefix)
+
+./script/build --install="${prefix}"
+for icon in resources/app-icons/stable/png/*.png
+do
+    size=$(basename "${icon}" | cut -d. -f1)
+    install -D "${icon}" "${prefix}/share/icons/hicolor/${size}x${size}/apps/io.atom.Atom.png"
```
Comment (lilic) [inline atom/install:11]: There is also `rename-icon`, which you can use in your json manifest file:
> Any icon with this name will be renamed to a name based on id during the cleanup phase.

**Review by lilic — CHANGES_REQUESTED** [review]
Will try building and let you know if I find anything else.

File: io.atom.Atom.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            "config-opts": [ "--disable-static", "--disable-gtk-doc", "--disable-coverage",
+                             "--disable-introspection" ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libgnome-keyring/3.12/libgnome-keyring-3.12.0.tar.xz",
+                    "sha256": "c4c178fbb05f72acc484d22ddb0568f7532c409b0a13e06513ff54b91e947783"
+                },
+                {
+                    "type": "patch",
```
Comment (lilic) [inline io.atom.Atom.json:52]: @krnowak I get the following error when building:
```
Downloading https://download.gnome.org/sources/libgnome-keyring/3.12/libgnome-keyring-3.12.0.tar.xz
Failed to download sources: module libgnome-keyring: Can't find file at libgnome-keyring/libgnome-keyring-detect-gcrypt.patch
```
Did you maybe forget to include the patch?

File: build.sh (hunk 1 above)
Comment (krnowak) [reply build.sh]: Haha, copy-pasta FTW.

File: atom/install (hunk 3 above)
Comment (krnowak) [reply atom/install:11]: Have to check it, but I'm not exactly renaming here. All the icons have the name like `<size>.png`, so `128.png` or `64.png`. But I'll check this `rename-icon` thing, thanks.

File: io.atom.Atom.json (hunk 4 above)
Comment (krnowak) [reply io.atom.Atom.json:52]: Duh, of course I forgot.

File: resources/install (hunk 2 above)
Comment (krnowak) [reply resources/install:8]: I would use it if I was going to only rename the desktop file, but I'm also patching it to use a different icon name, so I'll keep it like that for now.

File: atom/install (hunk 3 above)
Comment (krnowak) [reply atom/install:11]: Yeah, that wouldn't work for us here.

File: io.atom.Atom.json (hunk 4 above)
Comment (krnowak) [reply io.atom.Atom.json:52]: Fixed.

**Comment by krnowak** [comment]
PR updated.

**Comment by lilic** [comment]
Seems like this error repeats a lot:
```
Window manager warning: 0x1c00001 (untitled —) appears to be one of the offending windows with a timestamp of 36576935.  Working around...
Window manager warning: last_user_time (36577019) is greater than comparison timestamp (36577017).  This most likely represents a buggy client sending inaccurate timestamps in messages such as _NET_ACTIVE_WINDOW. Trying to work around...
```
Had some problems with running certain packages for example Hydrogen. Not sure if thats a flatpak error or the package error. Attaching screenshot.
screen shot 2017-03-08 at 17 10 33

One other thing is the init coffee script, css and the config files are empty, usually they come preloaded with comments and examples, not sure that works correctly there. Maybe you can have a look.

Otherwise works OK. 👍

**Comment by ptomato** [comment]
Do you know about https://github.com/mattdangerw/atom/commit/e54494fb63999174a7195b307dabd82dd7c0659f ?

File: io.atom.Atom.json [hunk 5]
```diff
@@ -0,0 +1,124 @@
+{
+    "app-id": "io.atom.Atom",
+    "base": "io.atom.electron.BaseApp",
+    "base-version": "master",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.4",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "atom",
+    "finish-args": [
+        /* Allow access to developer tools */
```
Comment (cosimoc) [inline io.atom.Atom.json]: Remove the comments, otherwise Jenkins will choke on the JSON.

**Review by cosimoc — CHANGES_REQUESTED** [review]

**Comment by krnowak** [comment]
@ptomato: Yes, I've seen that and I think this is a bit pointless - `./build/script --install=/app` does most of the things we want with fewer lines of code. Also, using flatpak-builder directly makes it easier to integrate with the endless jenkins instance.

File: io.atom.Atom.json (hunk 5 above)
Comment (krnowak) [reply io.atom.Atom.json]: Comments removed.

**Pull request merged**